test/third-party/follow-redirect-http.test.ts

```diff
@@ -92,3 +102,100 @@ it('intercepts a POST request issued by "follow-redirects"', async () => {
   expect(catchResponseUrl).toHaveBeenCalledWith(server.https.url('/user'))
   expect(await text()).toBe('hello from the server')
 })
+
+it('supports mocking a redirect response to the original response', async () => {
+  const requestListener = vi.fn<[Request]>()
+
+  interceptor.once('request', ({ request, controller }) => {
+    requestListener(request)
+
+    if (request.url.endsWith('/original')) {
+      controller.respondWith(
+        Response.redirect(server.https.url('/redirected'), 307)
+      )
+    }
+  })
+
+  const catchResponseUrl = vi.fn()
+  const request = https.request(
+    {
+      method: 'GET',
+      hostname: server.https.address.host,
+      port: server.https.address.port,
+      path: '/original',
+      rejectUnauthorized: false,
+    },
+    (res) => {
+      catchResponseUrl(res.responseUrl)
+    }
+  )
+
+  request.end()
+
+  const { text } = await waitForClientRequest(request as any)
+
+  // Intercepted redirect request (issued by "follow-redirects").
+  const [redirectedRequest] = requestListener.mock.calls[0]
+
+  expect(redirectedRequest.method).toBe('GET')
+  expect(redirectedRequest.url).toBe(server.https.url('/original'))
+
+  // Response (original).
+  expect(catchResponseUrl).toHaveBeenCalledWith(server.https.url('/redirected'))
+  expect(await text()).toBe('redirected response')
+})
+
+/**
+ * @note `follow-redirects` requires `http` and `https` before the interceptor
+ * can patch them.
+ */
+it.skip('supports mocking a redirect response to a mocked response', async () => {
```

Comment:
I'm skipping this scenario because it's import order-dependent. I wonder if this is what's causing Axios issues. Need to see if Axios uses the same third party to follow redirects (and how it even does that in XHR, which is the default request client there).

Reply:
Axios does use `follow-redirects`  - https://github.com/axios/axios/blob/5b8a826771b77ab30081d033fdba9ef3b90e439a/lib/adapters/http.js#L41

At least on Node by default. It uses the drop in replacements for the `http` and `https` modules provided by follow-redirects

Reply:
Nice! So that may be the root cause of redirects not working in Axios. I believe the http/https imports `follow-redirects` does are evaluated before the interceptors kick in. Need to verify that on an Axios test suite.